### DISCUSSION TITLE
Help with UDP handling

### DISCUSSION THREAD

**Discussion by sbonar** [libuv/libuv#4053] [Q&A] [answered]
New to libuv so I could be just missing something.  I have an app that is using several UDP sockets to read and write on and I have run into an issue where if I call either uv_udp_recv_stop() or uv_close() I get a SIGABRT in the run loop.
So the question is do I need to do something other than just closing the handle?

TIA

**Comment by bnoordhuis** [comment]
9 out of 10 times that's caused by people calling `uv_close()` and immediately freeing/reusing the handle, without waiting for the close callback first.

**Reply by sbonar** [reply]
Thank you.  A follow up question then does uv_close() clear the write queue so if there are outgoing buffers queued up they do not cause this issue?

**Reply by bnoordhuis** [reply, marked as answer]
Libuv invokes the callbacks of outstanding send/write requests with status == UV_ECANCELED before it calls the close callback.
